tslearn/tests/test_serialize_models.py

```diff
@@ -0,0 +1,253 @@
+
+    clear_tmp()
+
+
+def test_serialize_global_alignment_kernel_kmeans():
+    seed = 0
+    numpy.random.seed(seed)
+    X_train, y_train, X_test, y_test = CachedDatasets().load_dataset("Trace")
+    # Keep first 3 classes
+    X_train = X_train[y_train < 4]
+    numpy.random.shuffle(X_train)
+    # Keep only 50 time series
+    X_train = TimeSeriesScalerMeanVariance().fit_transform(X_train[:50])
+    sz = X_train.shape[1]
+
+    gak_km = GlobalAlignmentKernelKMeans(n_clusters=3,
+                                         sigma=sigma_gak(X_train),
+                                         n_init=20,
+                                         verbose=True,
+                                         random_state=seed)
+
+    _check_not_fitted(gak_km)
+
+    gak_km.fit(X_train)
+
+    _check_params_predict(gak_km, X_test)
+
+
+def test_serialize_timeserieskmeans():
+    seed = 0
+    numpy.random.seed(seed)
+    X_train, y_train, X_test, y_test = CachedDatasets().load_dataset("Trace")
+    X_train = X_train[y_train < 4]  # Keep first 3 classes
+    numpy.random.shuffle(X_train)
+    # Keep only 50 time series
+    X_train = TimeSeriesScalerMeanVariance().fit_transform(X_train[:50])
+    # Make time series shorter
+    X_train = TimeSeriesResampler(sz=40).fit_transform(X_train)
+    sz = X_train.shape[1]
+
+    dba_km = TimeSeriesKMeans(n_clusters=3,
+                              n_init=2,
+                              metric="dtw",
+                              verbose=True,
+                              max_iter_barycenter=10,
+                              random_state=seed)
+
+    _check_not_fitted(dba_km)
+
+    dba_km.fit(X_train)
+
+    _check_params_predict(dba_km, X_train)
+
+    sdtw_km = TimeSeriesKMeans(n_clusters=3,
+                               metric="softdtw",
+                               metric_params={"gamma_sdtw": .01},
+                               verbose=True,
+                               random_state=seed)
+
+    _check_not_fitted(sdtw_km)
+
+    sdtw_km.fit(X_train)
+
+    _check_params_predict(sdtw_km, X_train)
+
+
+def test_serialize_kshape():
+    seed = 0
+    numpy.random.seed(seed)
+    X_train, y_train, X_test, y_test = CachedDatasets().load_dataset("Trace")
+    # Keep first 3 classes
+    X_train = X_train[y_train < 4]
+    numpy.random.shuffle(X_train)
+    # Keep only 50 time series
+    X_train = TimeSeriesScalerMeanVariance().fit_transform(X_train[:50])
+    sz = X_train.shape[1]
+
+    ks = KShape(n_clusters=3, verbose=True, random_state=seed)
+
+    _check_not_fitted(ks)
+
+    ks.fit(X_train)
+
+    _check_params_predict(ks, X_train)
+
+
+def test_serialize_knn():
+    seed = 0
+    numpy.random.seed(seed)
+    X_train, y_train, X_test, y_test = CachedDatasets().load_dataset("Trace")
+
+    n_queries = 2
+    n_neighbors = 4
+
+    knn = KNeighborsTimeSeries(n_neighbors=n_neighbors)
+
+    _check_not_fitted(knn)
+
+    knn.fit(X_train)
+
+    # serialize to all all_formats
```

Comment:
Is it possible to use `_check_params_predict()` here or not?

Reply:
thanks for the suggestion, I generalized `_check_params_predict()` now.